lib/internal/fs/glob.js

```diff
@@ -376,6 +412,194 @@ class Glob {
       }
     }
   }
+
+
+  async* glob() {
```

Comment:
this is pretty much the same implementation as `globSync` except using `yield` instead if `this.#results.push`